### DISCUSSION TITLE
The __version__ attribute disappeared from version 5.0.1 to version 5.1.1

### DISCUSSION THREAD

**Discussion by ribes4** [miguelgrinberg/Flask-SocketIO#1671] [Q&A] [answered]
**Describe the bug**
Hi, I have been using `Flask-SocketIO` version 5.0.1 until today and when I queried the version with the `__version__` attribute it returned the current version as follows:

```
>>> import flask_socketio
>>> flask_socketio.__version__
'5.0.1'
```

But after upgrading the package to version 5.1.1 this attribute has disappeared:

```
>>> import flask_socketio
>>> flask_socketio.__version__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'flask_socketio' has no attribute '__version__'
```

**Expected behavior**
It was expected that the attribute would remain, if it has been moved, is there any way to query it again? 

Thanks!

**Comment by miguelgrinberg** [comment]
This attribute was never documented as a public attribute that applications can use. I have made some refactorings lately in the package and I don't really need it anymore.

**Comment by ribes4** [comment, marked as answer]
Thank you for your answer. Reading the PEP I have realized that this way of specifying the version became obsolete a few months ago (https://www.python.org/dev/peps/pep-0396/#pep-rejection).

And that a good way to check the version is using for example the `importlib` library. I write here the way to check the version, for future users:

For Python versions >=3.8:
```
>>> from importlib.metadata import version  
>>> version('flask_socketio')  
'5.1.1'
```
And for previous Python versions:

```
>>> from importlib_metadata import version
>>> version('flask_socketio')
'5.1.1'
```

**Comment by miguelgrinberg** [comment]
Oh, I didn't know that one!

I have always used pkg_resources as follows:

```python
>>> import pkg_resources
>>> pkg_resources.get_distribution("flask_socketio").version
'5.1.1'
```
